### ISSUE TITLE
Roboletric can't make custom dialog visible

### ISSUE THREAD

**Issue by gmmo** [robolectric/robolectric#4266]
### Description
I am stuck in a basically functionally I can't get it work with Roboletric. It is not making the fragment visibile on an custom android dialog.

### Steps to Reproduce
I posted the code to reproduce the issue on stackoverflow and it is pretty basic.
https://stackoverflow.com/questions/53619120/roboletric-cant-make-custom-dialog-visible

### Robolectric & Android Version
testImplementation 'org.robolectric:robolectric:4.0.2'

minSdkVersion 18
targetSdkVersion 28

### Link to a public git repo demonstrating the problem:
https://stackoverflow.com/questions/53619120/roboletric-cant-make-custom-dialog-visible

If anyone can answer on stackoverflow or give a hint on how to pass this wall I appreciate. I cannot write any unit-test for an android custom dialog if Roboletric can't make it visible.

thank you.

**Comment by paweljanowiak2** [comment]
I have similar issue with dialogs. It seems like dialog is not visible in layout hierarchy. When I copy test and run it on device everything is ok. 

Example: I have a custom spinner that on click opens a dialog that I can choose an option. 
```
spinner.performClick()
onView(withId(R.id.tv_dialog_title)).check(matches(isDisplayed())) // this fails
```

I've checked layout hierarchy and it is not there.

**Comment by mateot1** [comment]
Yeah it seems like Espresso matchers/actions on Dialogs/DialogFragments don't work with Robolectric... For a visibility check you can do something like assert that `ShadowDialog.getLatestDialog().isShowing == true` but I'm interesting in knowing if there's a recommendation for performing click actions or assertions on subviews of the Dialog.

**Comment by mateot1** [comment]
oh wait I just looked a little more and found the `clickOn` method... so you can do something like 
```
shadowOf(ShadowDialog.getLatestDialog()).clickOn(android.R.id.button1) // click on the default positive button
``` 

But I do think it would be ideal to be able to use espresso actions since it seems like the plan is to merge the apis
